sstables/sstables.hh

```diff
@@ -948,6 +949,15 @@ class file_io_extension {
 public:
     virtual ~file_io_extension() {}
     virtual future<file> wrap_file(sstable&, component_type, file, open_flags flags) = 0;
+    // optionally return a map of attributes for a given sstable,
+    // suitable for "describe".
+    // This would preferably be interesting info on what/why the extension did
+    // to this table.
+    using attr_value_type = std::variant<sstring, std::map<sstring, sstring>>;
+    using attr_value_map = std::map<sstring, attr_value_type>;
+    virtual attr_value_map get_attributes(const sstable&) const {
+        return {};
+    }
 };
```

Comment:
I think it's nicer to return C++ structs here and not unstructured data, then do the conversion at the API level.

Probably not worth the effort.

Reply:
How do you create a struct to describe a property map that is not, you know, a ... map...?
The point is to return arbitrary key-values (with value optionally being another key-value set).

Reply:
They're not arbitrary.

```c++
struct sstable_attributes {
    bool is_compressed;
    sstring creator_name;
};
```

it's kind of pointless in this case because the only consumer will mangle them to a map, but that's how I would have done it, I like delaying the destruction of structure as far as I can.

Reply:
Those attributes are not handled by this. This is only for extensions to provide info not known to the collecting API call. 
All "known" quantities are handled structured. 
This is basically for E@R and other extensions to be able to provide a set of properties.

Reply:
they could provide proper methods too (but exposing complex attributes is hard)